carbonmark/components/pages/Users/SellerConnected/index.tsx

```diff
@@ -124,8 +125,13 @@ export const SellerConnected: FC<Props> = (props) => {
       // Merge with data from Updated Profile as backend might be slow!
       const newUser = { ...userFromApi, ...profileData };
 
-      // Update the cache only, do not revalidate
+      // Update the cache only, this is an optimistic update
       await mutate(newUser, false);
+      //Update the server directly. This is a temporary solution until we can clean up the mutation logic
+      mutate(handleMutateUserUntil, {
```

Comment:
if we have the newUser data already why do we need to poll? afaik that poller would only end when new activity is found but `onEditProfile` doesn't create an activity entry

Reply:
If you remove this chunk, update your profile and then switch pages via the menu and go back to the profile page then the changes aren't persisted. It seems to pull the previous profile value from the cache.

To be fair the wait until logic confused me significantly so I just stole this from where the server mutation as being done elsewhere. Maybe you have a simpler solution?

Reply:
@0xemc I think you just need to add populateCache:true to line 129, did that not work?

Reply:
I thought the same but then the UI is not updating.

There is something extremely wacky going on with the updating of users. Updating the profile submits a POST request which is immediately followed by 4 GET calls that override the update. I've tried using optimistic data as well but it seems to get overridden by the subsequent GET.

<img width="838" alt="Screenshot 2023-07-26 at 9 38 59 am" src="https://github.com/KlimaDAO/klimadao/assets/7104689/5d5f10ca-d604-4dae-9a75-19aa2f475863">

I'm loathe to refactor how this is working and spend too much time on it so for now I thought I'd replicate how it was done elsewhere